lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb

```diff
@@ -71,7 +71,18 @@ def get_last_milestone(repository)
         last_stone
       end
 
-      def create_milestone(repository, newmilestone_number, newmilestone_duedate, newmilestone_duration, number_of_days_from_code_freeze_to_release, need_submission)
+      # Creates a new milestone
+      #
+      # @param [String] repository The repository name, including the organization (e.g. `wordpress-mobile/wordpress-ios`)
+      # @param [String] newmilestone_number The name of the milestone we want to create (e.g.: `16.9`)
+      # @param [Time] newmilestone_duedate milestone due date (e.g. `2022-10-22T12:00:00Z`)
+      # @param [Integer] newmilestone_duration Number of days that a milestone extents
+      # @param [Integer] number_of_days_from_code_freeze_to_release Number of days from code freeze to release
+      # @param [Boolean] need_submission The app needs to be submitted?
+      #  if `true`, will subtract 3 days from the `:number_of_days_from_code_freeze_to_release`.
+      #  if `false`, will use the days of `:newmilestone_duration`
+      #
+      def create_milestone(repository:, newmilestone_number:, newmilestone_duedate:, newmilestone_duration:, number_of_days_from_code_freeze_to_release:, need_submission:)
```

Comment:
The parameter names seems a bit of a mouthful to me, given we already know from the context (of them being parameters of `create_milestone`) that those are about new milestones.

For example, what about naming the parameters just `title`, `due_date`, `duration`, and `days_until_release`?

Though a probably even nicer solution, especially because `duration` and `need_submission` are only used once and are kind of exclusive (i.e. either we pass `need_submission: false` and then `duration` doesn't matter, or we pass `need_submission: true` and that's only when `duration` is used…), we should probably combine the goal of the two parameters into a single one there, which would be `days_until_submission`, and let the caller pass to this value.

Given that, I suggest the following new API for this helper method:

```ruby
      def create_milestone(repository:, title:, due_date:, days_until_submission:, days_until_release:)
```

Then:
 - For the time being, the call site in `create_new_milestone_action` would just do the computation on its end, i.e. pass `params[:need_appstore_submission] ? number_of_days_from_code_freeze_to_release - 3 : milestone_duration` for the new `days_until_submission` parameter of this `GithubHelper#create_milestone`, carrying over the "Using 3 days mostly for historical reasons" logic from lines 88-89 for now
 - In the future, when we will [consider updating the API of the `create_new_milestone` action itself](https://github.com/wordpress-mobile/release-toolkit/pull/425#discussion_r1014337960) one day, we would then be able to update its `ConfigItem` (in a breaking-change way) to have the callers also directly provide the `days_until_submission` and `days_until_release` values in their `Fastfile`'s call sites instead of `duration` and/or `need_submission`

But for now, that change would make for a nicer and easier to read and understand API for the internal `GithubHelper#create_milestone`, all while preparing for the future of the planned refactor of the corresponding action and public API.

Reply:
For sure, those parameters that are in place are big! 
I'll address those changes, thanks for the input 😄